Sources/Common/Concurrency/TaskExtension.swift

```diff
@@ -18,51 +18,72 @@
 
 import Foundation
 
+public struct Sleeper {
+
+    public static let `default` = Sleeper(sleep: {
+        try await Task<Never, Never>.sleep(interval: $0)
+    })
+
+    private let sleep: (TimeInterval) async throws -> Void
+
+    public init(sleep: @escaping (TimeInterval) async throws -> Void) {
+        self.sleep = sleep
```

Comment:
Should we guard against negative values here?

Reply:
no, it will crash in that case